playbooks/besman-sonar-scanner-0.0.1-steps.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Author: Samir Ranjan Parhi
+# License: Same as Repository Licence
+# Usage : Beta-Release
+# Date : 12/02/2024
+
+    echo "Running $ASSESSMENT_TOOL_NAME"
+     cd "$BESMAN_TOOL_PATH" || return 1
+
+    export SONAR_RESULT=1
+
+    # Check if Docker is installed
+    if ! command -v docker &> /dev/null; then
+        echo -e "\n ❌ Docker is not installed. Please install Docker before running this script."
+        SONAR_RESULT=0
+        exit 1
+        
+    fi
+
+    # Check if SonarQube container is running
+
+    if docker ps -a --format '{{.Names}}' | grep -Eq '^sonarqube$'; then
+        echo -e "\n✅ SonarQube container is already running."
+    else
+        echo -e "\n🛫 Starting SonarQube container..."
+        docker run -d --name sonarqube -p 9000:9000 -p 9092:9092 sonarqube
+    fi
+    if ! command -v sonar-scanner &> /dev/null; then
+        echo -e "\n❌ Sonar Scanner CLI is not installed. Please install Sonar Scanner CLI before running this script."
+        SONAR_RESULT=0
+        exit 1
+    fi
+
+    sonar-scanner \
+    -Dsonar.projectKey=$BESMAN_ARTIFACT_NAME \
+    -Dsonar.projectName="$BESMAN_ARTIFACT_NAME" \
+    -Dsonar.host.url=$SONAR_HOST_URL \
```

Comment:
Where is this variable set?

Reply:
@asa1997 : variable is defined in the playbook init section

Reply:
```
local var_array=("BESMAN_ARTIFACT_TYPE" "BESMAN_ARTIFACT_NAME" "BESMAN_ARTIFACT_VERSION" "BESMAN_ARTIFACT_URL" "BESMAN_ENV_NAME" "BESMAN_ARTIFACT_DIR" "ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_VERSION" "ASSESSMENT_TOOL_PLAYBOOK" "BESMAN_ASSESSMENT_DATASTORE_DIR" "BESMAN_TOOL_PATH" "BESMAN_ASSESSMENT_DATASTORE_URL" "BESMAN_LAB_OWNER_TYPE" "BESMAN_LAB_OWNER_NAME" "SONAR_HOST_URL" "SONAR_LOGIN_TOKEN")
```
is this the right place